addons/pos_coupon/static/src/js/OrderWidget.js

```diff
@@ -6,6 +6,9 @@ odoo.define('pos_coupon.OrderWidget', function (require) {
 
     const PosCouponOrderWidget = (OrderWidget) =>
         class PosCouponOrderWidget extends OrderWidget {
+            setup() {
+                this.activeProgramsProps = getActiveProgramsProps();
+            }
```

Comment:
Don't do this. You should do this assignment in the template because when OrderWidget updated, this setup won't be ran again. However, the template will be recalculated.

Reply:
I wanted to avoid calling `getActiveProgramsProps()` four times. But you say it actually has to be calculated four times, right?

Reply:
you can use `t-set` directive to save the result.
check my comment here: https://github.com/odoo/odoo/pull/84317#discussion_r806808851